modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/IgniteTestUtils.java

```diff
@@ -722,12 +725,55 @@ public static String randomString(Random rnd, int len) {
         return sb.toString();
     }
 
+    /**
+     * Returns random {@link BigDecimal} with given scale and precision.
+     *
+     * @param rnd Random generator.
+     * @param scale Scale of generating value.
+     * @param precision Precision of generating value.
+     *
+     * @return Random string.
+     */
+    public static BigDecimal randomBigDecimal(Random rnd, int scale, int precision) {
+        assert precision >= scale;
+        assert precision > 0;
```

Comment:
the same, this is general purpose utils, why do it has SQL-imposed limitations? Besides, this limitation is for type declaration, but the actual values given by user may not conform to this

Reply:
Agree